packages/plugins/typed-variable-modal/test/typed_variable_modal_test.mocha.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Unit tests for TypedVariableModal.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+const assert = require('assert');
+const Blockly = require('blockly');
+const sinon = require('sinon');
+
+const TypedVariableModal = require('../dist/index.js').TypedVariableModal;
+
+suite('TypedVariableModal', () => {
+  /**
+   * Set up the workspace to test with typed variable modal.
+   * @param {string} toolbox The toolbox.
+   * @param {Array.<Array.<string>>} types An array holding arrays with the
+   *     display name as the first value and the type as the second.
+   * @return {Blockly.WorkspaceSvg} The workspace to use for testing.
+   */
+  function workspaceSetup(toolbox, types) {
+    const options = {};
+    const createFlyout = function(workspace) {
+      let xmlList = [];
+      const button = document.createElement('button');
+      button.setAttribute('text', 'Create Typed Variable');
+      button.setAttribute('callbackKey', 'CREATE_TYPED_VARIABLE');
+
+      xmlList.push(button);
+
+      const blockList = Blockly.VariablesDynamic
+          .flyoutCategoryBlocks(workspace);
+      xmlList = xmlList.concat(blockList);
+      return xmlList;
+    };
+    options.toolbox = toolbox;
+
+    const ws = Blockly.inject('blocklyDiv', options);
+    ws.registerToolboxCategoryCallback('CREATE_TYPED_VARIABLE', createFlyout);
+    return ws;
+  }
+  /**
+   * Create a toolbox to test with.
+   * @return {string} The toolbox.
+   */
+  function getTestToolbox() {
+    const toolbox = `
+      <xml>
+        <category name="Typed Variables" categorystyle="variable_category"
+        custom="CREATE_TYPED_VARIABLE"></category>
+      </xml>
+    `;
+    return toolbox;
+  }
+
+  setup(() => {
+    this.jsdomCleanup =
+        require('jsdom-global')('<!DOCTYPE html><div id="blocklyDiv"></div>');
+    const types = [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']];
+    this.workspace = workspaceSetup(getTestToolbox(), types);
+    this.typedVarModal = new TypedVariableModal(this.workspace,
+        'CREATE_TYPED_VARIABLE', types);
+  });
+
+  teardown(() => {
+    this.jsdomCleanup();
+    sinon.restore();
+  });
+
+  suite('init()', () => {
+    test('Render is called', () => {
+      this.workspace.registerButtonCallback = sinon.fake();
+      this.typedVarModal.init();
+      assert(this.workspace.registerButtonCallback.calledOnce);
```

Comment:
Roughly looking at documentation, it seems like you could do:
`sinon.assert.calledOnce(this.workspace.registerButtonCallback);`
Assuming it works, I think it might give better error messages.